services/consensuscontext/create_transactions_block.go

```diff
@@ -48,10 +53,25 @@ func (s *service) createTransactionsBlock(ctx context.Context, input *services.R
 			NumSignedTransactions:      uint32(txCount),
 		}).Build(),
 		Metadata:           metaData,
-		SignedTransactions: proposedTransactions.SignedTransactions,
+		SignedTransactions: transactionsForBlock,
 		BlockProof:         (&protocol.TransactionsBlockProofBuilder{}).Build(),
 	}
 
 	s.metrics.transactionsRate.Measure(int64(len(txBlock.SignedTransactions)))
 	return txBlock, nil
 }
+
+func (s *service) updateTransactions(txs []*protocol.SignedTransaction, blockTime primitives.TimestampNano) []*protocol.SignedTransaction {
+	if s.config.ConsensusContextTriggersEnabled() {
+		txs = append(txs, (&protocol.SignedTransactionBuilder{
```

Comment:
This would be much nicer if you wrapped the trigger tx initialization in a function